src-qt5/core/libLumina/LuminaOS-NixOS.cpp

```diff
@@ -0,0 +1,315 @@
+#include "LuminaOS.h"
+#include <unistd.h>
+#include <stdio.h> // Needed for BUFSIZ
+
+//can't read xbrightness settings - assume invalid until set
+static int screenbrightness = -1;
+
+QString LOS::OSName(){ return "NixOS"; }
+
+//OS-specific prefix(s)
+// NOTE: PREFIX, L_ETCDIR, L_SHAREDIR are defined in the OS-detect.pri project file and passed in
+QString LOS::LuminaShare(){ return (L_SHAREDIR+"/lumina-desktop/"); } //Install dir for Lumina share files
+QString LOS::AppPrefix(){ return "/run/current-system/sw"; } //Prefix for applications
+QString LOS::SysPrefix(){ return "/run/current-system/sw"; } //Prefix for system
+
+//OS-specific application shortcuts (*.desktop files)
+QString LOS::ControlPanelShortcut(){ return "lumina-config.desktop"; } //system control panel
+QString LOS::AppStoreShortcut(){ return ""; } //graphical app/pkg manager
+//OS-specific RSS feeds (Format: QStringList[ <name>::::<url> ]; )
+QStringList LOS::RSSFeeds(){ return QStringList(); }
+
+// ==== ExternalDevicePaths() ====
+QStringList LOS::ExternalDevicePaths(){
+  /* Returns: QStringList[<type>::::<filesystem>::::<path>]
+     Note: <type> = [USB, HDRIVE, DVD, SDCARD, UNKNOWN]
+     df is much better for this than mount, because it skips
+     all non-physical devices (like bind-mounts from schroot)
+     so they are not listed in lumina-fm */
+  QStringList devs = LUtils::getCmdOutput("@df@ --output=source,fstype,target");
+  //Now check the output
+  for(int i=0; i<devs.length(); i++){
+    if(devs[i].startsWith("/dev/")){
+      devs[i] = devs[i].simplified();
+      QString type = devs[i].section(" on ",0,0);
+      type.remove("/dev/");
+      //Determine the type of hardware device based on the dev node
+      if(type.startsWith("sd") || type.startsWith("nvme")){ type = "HDRIVE"; }
+      else if(type.startsWith("sr")){ type="DVD"; }
+      else if(type.contains("mapper")){ type="LVM"; }
+      else{ type = "UNKNOWN"; }
+      //Now put the device in the proper output format
+      devs[i] = type + "::::" + devs[i].section(" ",1,1) + "::::" + devs[i].section(" ",2,2);
+    }else{
+      //invalid device - remove it from the list
+      devs.removeAt(i);
+      i--;
+    }
+  }
+  return devs;
+}
+
+//Read screen brightness information
+int LOS::ScreenBrightness(){
+   //Returns: Screen Brightness as a percentage (0-100, with -1 for errors)
+  if(screenbrightness==-1){
+    if(QFile::exists(QString(getenv("XDG_CONFIG_HOME"))+"/lumina-desktop/currentxbrightness")){
+      int val = LUtils::readFile(QString(getenv("XDG_CONFIG_HOME"))+"/lumina-desktop/currentxbrightness").join("").simplified().toInt();
+      screenbrightness = val;
+    }
+  }
+  return screenbrightness;
+}
+
+//Set screen brightness
+void LOS::setScreenBrightness(int percent){
+  //ensure bounds
+  if(percent<0){percent=0;}
+  else if(percent>100){ percent=100; }
+  // float pf = percent/100.0; //convert to a decimel
+  //Run the command
+  QString cmd = "@brightnessctl@ set %1%%";
```

Comment:
@SuperSandro2000 I dont believe there would be any difference in how variables work on Nix, but I figured I'd check. @romildo asked me if the 2nd % at the end of this string was needed.  I dont think it is, since `%1` is the variable and `%` is the symbol.  But I figured before I merged this in I'd check in case there was some odd thing abut Nix I wasnt aware of with how it would handle this. I dont think there would be, but I wanted to make sure.
```suggestion
  QString cmd = "@brightnessctl@ set %1%";
```

Reply:
I can't tell you anything about that but the @variables@ are replaced by nix with the actual paths.

Reply:
The [documentation for `QString::arg`](https://doc.qt.io/qt-5/qstring.html#arg) does not mention `%%`. So it may be secure to assume it has no special meaning. To have a single `%` in the string it would not be needed to write `%%` in the literal string.